### PR TITLE
Support for Markdown javadoc (JEP-467)

### PR DESCRIPTION
**Pull request by lahodaj**
A patch to support Markdown javadoc (JEP-467).

### CODE REVIEW

File: java/java.sourceui/src/org/netbeans/api/java/source/ui/ElementJavadoc.java [hunk 1]
```diff
@@ -1909,6 +1965,9 @@ public SourceLineCharterMapperToHtmlTag(List<String> startTag, char sourceChar,
         }
     }
     private void appendReference(StringBuilder sb, ReferenceTree ref, List<? extends DocTree> label, TreePath docPath, DocCommentTree doc, DocTrees trees) {
+        if (ref == null) {
+            System.err.println("!!!!");
+        }
```
Comment (mbien) [inline java/java.sourceui/src/org/netbeans/api/java/source/ui/ElementJavadoc.java:1970]: `System.err` used outside of tests

File: java/java.editor.base/test/unit/src/org/netbeans/modules/java/editor/base/javadoc/JavadocImportsTest.java [hunk 2]
```diff
@@ -207,6 +207,122 @@ public void testComputeReferencedElements() throws Exception {
         assertEquals(exp, sortedResult);
     }
     
+    public void testComputeReferencedElementsMarkdown() throws Exception {
+        String code =
+                "package p;\n" +
+                "import java.io.IOException;\n" +
```
Comment (mbien) [inline java/java.editor.base/test/unit/src/org/netbeans/modules/java/editor/base/javadoc/JavadocImportsTest.java:213]: sidenote: bumping the lang level for multi line Strings could make it easier to write new tests with code snippets.

**Review by dbalek — APPROVED** [review]
Looks fine to me. Thanks.

**Comment by lahodaj** [comment]
Unless there are objections, I will merge in the next day or so.

**Comment by mbien** [comment]
probably ok to merge as new feature, but I fear that this might confuse some of the javadoc editor hints since they probably don't expect markdown.

**Comment by lahodaj** [comment]
> probably ok to merge as new feature, but I fear that this might confuse some of the javadoc editor hints since they probably don't expect markdown.

It might confuse something. Or some piece of code may be confused by the current situation (where the model already includes AST model of Markdown javadoc, but the JavadocUtilities don't understand it). I don't think this is an actionable claim, and I think this change is a step in finding out what's wrong, if anything, and fix it.

**Comment by mbien** [comment]
> I don't think this is an actionable claim

it was just a feeling, but here an example after testing it quickly:

```java
    ///
    /// @param str
    ///
    public static void foo(String str, int i) {}
```
the hint will suggest to insert the doc for `i` in the old format, not knowing that the method has already a block in the new format.

result:
```java
    ///
    /// @param str
    ///

    /**
     *
     * @param str
     * @param i
     */
    public static void foo(String str, int i) {}
```


> and I think this change is a step in finding out what's wrong, if anything, and fix it.

of course. That is why I said I think it is ok to merge this as new feature.

**Review by mbien — APPROVED** [review]
forgot to approve - looks good!

**Pull request merged**